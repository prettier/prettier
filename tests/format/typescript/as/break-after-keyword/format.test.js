runFormatTest(import.meta, ["typescript", "flow"], {
  errors: { hermes: ["18148.ts"] },
});
