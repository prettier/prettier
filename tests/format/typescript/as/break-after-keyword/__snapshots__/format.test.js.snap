// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`18148.ts [hermes] format 1`] = `
"';' expected (12:70)
  10 |
  11 | // 79 width until \`as\` or \`satisfies\` keyword
> 12 | const firstItem3 = _____________.find((item) => item.type === 'item') satisfies string | number;
     |                                                                      ^
  13 | // 80 width until \`as\` or \`satisfies\` keyword
  14 | const firstItem_3 = _____________.find((item) => item.type === 'item') satisfies string | number;
  15 |
Cause: ';' expected (12:70)
const firstItem3 = _____________.find((item) => item.type === 'item') satisfies string | number;
                                                                      ^"
`;

exports[`18148.ts format 1`] = `
====================================options=====================================
parsers: ["typescript", "flow"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// 79 width until \`as\` or \`satisfies\` keyword
const firstItem1 = ______________.items.find((item) => item.type === 'item') as const;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_1 = ______________.items.find((item) => item.type === 'item') as const;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem2 = ______________.items.find((item) => item.type === 'item') as string | number;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_2 = ______________.items.find((item) => item.type === 'item') as string | number;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem3 = _____________.find((item) => item.type === 'item') satisfies string | number;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_3 = _____________.find((item) => item.type === 'item') satisfies string | number;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem4 = ______________.items.find((item) => item.type === 'item') as not_union;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_4 = ______________.items.find((item) => item.type === 'item') as not_union;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem5 = ______________.items.find((item) => item.type === 'item') as a_union_will_break | // comments
a_union_will_break2;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_5 = ______________.items.find((item) => item.type === 'item') as a_union_will_break | // comments
a_union_will_break2;

=====================================output=====================================
// 79 width until \`as\` or \`satisfies\` keyword
const firstItem1 = ______________.items.find(
  (item) => item.type === "item",
) as const;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_1 = ______________.items.find(
  (item) => item.type === "item",
) as const;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem2 = ______________.items.find((item) => item.type === "item") as
  | string
  | number;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_2 = ______________.items.find(
  (item) => item.type === "item",
) as string | number;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem3 = _____________.find((item) => item.type === "item") satisfies
  | string
  | number;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_3 = _____________.find(
  (item) => item.type === "item",
) satisfies string | number;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem4 = ______________.items.find(
  (item) => item.type === "item",
) as not_union;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_4 = ______________.items.find(
  (item) => item.type === "item",
) as not_union;

// 79 width until \`as\` or \`satisfies\` keyword
const firstItem5 = ______________.items.find((item) => item.type === "item") as
  | a_union_will_break // comments
  | a_union_will_break2;
// 80 width until \`as\` or \`satisfies\` keyword
const firstItem_5 = ______________.items.find(
  (item) => item.type === "item",
) as
  | a_union_will_break // comments
  | a_union_will_break2;

================================================================================
`;
